library/signed/src/RippleCarry.qs

```diff
@@ -0,0 +1,418 @@
+///   https://arxiv.org/abs/0910.2530
+///
+/// # Remarks
+/// The specified controlled operation makes use of symmetry and mutual
+/// cancellation of operations to improve on the default implementation
+/// that adds a control to every operation.
+operation ApplyInnerTTKAdder(xs : Qubit[], ys : Qubit[], carry : Qubit) : Unit is Adj + Ctl {
+    body (...) {
+        (Controlled ApplyInnerTTKAdder)([], (xs, ys, carry));
+    }
+    controlled (controls, ...) {
+        let nQubits = Length(xs);
+
+        for idx in 0..nQubits - 2 {
+            CCNOT(xs[idx], ys[idx], xs[idx + 1]);
+        }
+        (Controlled CCNOT)(controls, (xs[nQubits-1], ys[nQubits-1], carry));
+        for idx in nQubits - 1..-1..1 {
+            Controlled CNOT(controls, (xs[idx], ys[idx]));
+            CCNOT(xs[idx-1], ys[idx-1], xs[idx]);
+        }
+    }
+}
+
+/// # Summary
+/// Implements the outer operation for RippleCarryAdderTTK to conjugate
+/// the inner operation to construct the full adder.
+///
+/// # Input
+/// ## xs
+/// LittleEndian qubit register encoding the first integer summand
+/// input to RippleCarryAdderTTK.
+/// ## ys
+/// LittleEndian qubit register encoding the second integer summand
+/// input to RippleCarryAdderTTK.
+///
+/// # References
+/// - Yasuhiro Takahashi, Seiichiro Tani, Noboru Kunihiro: "Quantum
+///   Addition Circuits and Unbounded Fan-Out", Quantum Information and
+///   Computation, Vol. 10, 2010.
+///   https://arxiv.org/abs/0910.2530
+operation ApplyOuterTTKAdder(xs : Qubit[], ys : Qubit[]) : Unit is Adj + Ctl {
+    let nQubits = Length(xs);
+
+    Fact(
+        nQubits == Length(ys),
+        "Input registers must have the same number of qubits."
+    );
+
+    ApplyToEachCA(CNOT, Zipped(Rest(xs), Rest(ys)));
+    Adjoint ApplyCNOTChain(Rest(xs));
+}
+
+/// # Summary
+/// Reversible, in-place ripple-carry addition of two integers.
+/// Given two $n$-bit integers encoded in LittleEndian registers `xs` and `ys`,
+/// and a qubit carry, the operation computes the sum of the two integers
+/// where the $n$ least significant bits of the result are held in `ys` and
+/// the carry out bit is xored to the qubit `carry`.
+///
+/// # Input
+/// ## xs
+/// LittleEndian qubit register encoding the first integer summand.
+/// ## ys
+/// LittleEndian qubit register encoding the second integer summand, is
+/// modified to hold the $n$ least significant bits of the sum.
+/// ## carry
+/// Carry qubit, is xored with the most significant bit of the sum.
+///
+/// # References
+/// - Yasuhiro Takahashi, Seiichiro Tani, Noboru Kunihiro: "Quantum
+///   Addition Circuits and Unbounded Fan-Out", Quantum Information and
+///   Computation, Vol. 10, 2010.
+///   https://arxiv.org/abs/0910.2530
+///
+/// # Remarks
+/// This operation has the same functionality as RippleCarryAdderD and,
+/// RippleCarryAdderCDKM but does not use any ancilla qubits.
+operation RippleCarryAdderTTK(xs : Qubit[], ys : Qubit[], carry : Qubit) : Unit is Adj + Ctl {
```

Comment:
A few functions are already here: https://github.com/microsoft/qsharp/blob/9ae1d2a1ae9c4a885b88d2586302ca53555b5e0f/library/unstable/src/Arithmetic.qs#L159

Reply:
Thanks, given the naming differences I wasn't sure if those are actually the same function

Reply:
I've removed the port of the old ripple carry and am using `library/unstable` now. Thanks!